docs/src/architecture/state.md

```diff
@@ -1,74 +1,86 @@
 
-`hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`.
+### Account database
+The latest account states are recorded in a [tiered sparse Merkle tree](https://0xpolygonmiden.github.io/miden-base/crypto-primitives/tsmt.html) which maps account IDs to account hashes and account data if needed. 
 
 <p align="center">
-  <img src="../diagrams/architecture/state/Account_DB.png">
+  <img src="../diagrams/architecture/state/Account_DB.png" width="80%">
 </p>
 
-As described in [accounts](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html), there are two types of accounts:
+As described in [Accounts](http://localhost:3000/architecture/accounts.html#account-storage-modes), there are three types of accounts:
 * **Public accounts** where all account data is stored on-chain.
+* **Encrypted accounts** where where the account data is stored onchain but in encrypted text.
 * **Private accounts** where only the hashes of accounts are stored on-chain.
 
-> Losing a state of a private account would mean loss of funds (as the user won't be able to execute transactions) in a similar manner as a loss of a private key would. This problem can be easily mitigated by storing encrypted account state in a cloud or backing it up somewhere else. Unlike storing private keys in the cloud, this does not compromise privacy or security of an account.
+Having many (or even most) of the accounts be private is very beneficial for the network as a private account contributes only $40$ bytes to the global state ($8$ bytes account ID + $32$ bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.
 
-Note: Having many (or even most) of the accounts be private is very beneficial for the network as a private account contributes only 40 bytes to the global state (8 bytes account ID + 32 bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.
+> Losing a state of a private account would mean loss of funds (as the user won't be able to execute transactions) in a similar manner as a loss of a private key would. This problem can be easily mitigated by storing encrypted account state in a cloud or backing it up somewhere else. Unlike storing private keys in the cloud, this does not compromise privacy or security of an account.
 
 ### Note database
+Notes are recorded in an append-only accumulator, a [Merkle Mountain Range](https://github.com/opentimestamps/opentimestamps-server/blob/master/doc/merkle-mountain-range.md). Each leaf is a block header which contains the commitment to all notes created in that block. Note, the size of the Merkle Mountain Range grows logarithmically with the number of items in it.
```

Comment:
I would remove "Note, " from the sentence here:

```
Note, the size of the Merkle Mountain Range grows logarithmically with the number of items in it.
```

It's overloaded with the object Note that we are discussing.

Reply:
hehe you are right